### PR TITLE
Initial support for Bit 0

### PR DESCRIPTION
**Pull request by gameboo**

### CODE REVIEW

File: Haskell/Blarney/Core/Net.hs [hunk 1]
```diff
@@ -245,6 +247,80 @@ inlineSingleRefNet nl = do
...
+isZeroWidthRootNet _ = False
+
+-- | Ignore 0-width Nets pass
+ignoreZeroWidthNet :: MNetlist -> IO Bool
+ignoreZeroWidthNet nl = do
+  pairs <- getAssocs nl -- list of nets with their index
+  changed <- newIORef False -- keep track of modifications to the 'Netlist'
+  -- For each 'Net' (in particular, those with a non-zero-width output) with at
+  -- least a zero-width input, transform it into a 'Net' with no reference to
+  -- any other zero-width output Net.
```
Comment (mn416) [inline Haskell/Blarney/Core/Net.hs]: For each 'Net' (in particular, those with a non-zero-width output) with at
least **one** zero-width input, transform it into a 'Net' with no reference to
any ~~other~~ zero-width output Net.

File: Haskell/Blarney/Core/Net.hs [hunk 2]
```diff
@@ -245,6 +247,80 @@ inlineSingleRefNet nl = do
...
+--zeroWidthNetTransform net@Net{ netPrim = BRAM {} } =
+--  error "zeroWidthNetTransform unsupported on BRAM Prim"
+--zeroWidthNetTransform net@Net{ netPrim = TrueDualBRAM {} } =
+--  error "zeroWidthNetTransform unsupported on TrueDualBRAM Prim"
+--zeroWidthNetTransform net@Net{ netPrim = Custom {} } =
+--  error "zeroWidthNetTransform unsupported on Custom Prim"
+--zeroWidthNetTransform net@Net{ netPrim = RegFileRead {} } =
+--  error "zeroWidthNetTransform unsupported on RegFileRead Prim"
+-- do nothing cases for all others
+zeroWidthNetTransform net = (net, False)
```
Comment (mn416) [inline Haskell/Blarney/Core/Net.hs]: Eliminate `countOnes` prim from library -- we can implement within Blarney.

BRAM, TrueDualBRAM, RegFileRead --> Const 0, when addr width is 0.

Prefer Const 0 to DontCare.

**Comment by gameboo** [comment]
should be good to go now?

**Comment by mn416** [comment]
Can we have a contrived example that demonstrates working behaviour for:

  * Equality with 0-width inputs
  * Bit concatenation with a 0-width input
  * Zero or sign extension of 0-width input
  * Bit selection (actually, do we need to process these explicitly?)
  * RAM with 0-width address
  * Display with a "%d" of a 0-width value

**Comment by mn416** [comment]
Looking good.  Can we drop the `SelectBits` case and fill in the missing `Custom` case?

**Comment by mn416** [comment]
Looks good. Can’t see any issues.

**Comment by mn416** [comment]
Does your AXI library go through ok?

**Pull request merged**